src/installer/corehost/cli/apphost/static/CMakeLists.txt

```diff
@@ -131,11 +133,24 @@ else()
     set(CORECLR_LIBRARIES
         ${CORECLR_STATIC_LIB_LOCATION}/libcoreclr_static.a
         ${CORECLR_STATIC_LIB_LOCATION}/libSystem.Globalization.Native.a
+        ${CORECLR_STATIC_LIB_LOCATION}/libSystem.IO.Compression.Native.a
         ${CORECLR_STATIC_LIB_LOCATION}/libpalrt.a
         ${CORECLR_STATIC_LIB_LOCATION}/libcoreclrpal.a
         ${CORECLR_STATIC_LIB_LOCATION}/libeventprovider.a
         ${CORECLR_STATIC_LIB_LOCATION}/libnativeresourcestring.a
     )
+
+    # additional requirements for System.IO.Compression.Native
```

Comment:
This is rather fragile. Can't these propagate from the native libs themselves?

Reply:
It would be nice to have this in one place per-library. It gets worse with cryptography libraries and extra requirements could be quite ad-hoc, not necessary just extra libs.
I'll see what can be done.

Reply:
I have extracted parts that deal with additional libs into separate cmake files and use in all places that need to link the corresponding lib (right now it is when building .so or when building static host).

This way, if a new requirement comes up, we would need to make a change in only one place.